src/Common/Entity/Transaction.php

```diff
@@ -202,31 +202,35 @@ public function setProducts($products)
         $this->products = $container;
     }
 
+    public function getTransactionSource(): string
+    {
+        return $this->transaction_source;
+    }
+
+    public function setTransactionSource(string $transaction_source): void
+    {
+        $this->transaction_source = $transaction_source;
+    }
+
     public function getSourceDisplay()
     {
         return ucwords(strtolower(str_replace('_', ' ', $this->getSource())));
     }
 
     /**
+     * @NOTE: I suppose we could move this, but that would break backwards compatibility for no real benefit
      * @return string
      */
     public function getSource(): string
     {
-        foreach ($this->getMeta() as $meta) {
-            $data = (array)$meta;
-            if (empty($data['TRANSACTION_SOURCE']) === false) {
-                return $data['TRANSACTION_SOURCE'];
-            }
+        if (isset($this->transaction_source)) {
```

Comment:
I assume we'll chase this update with an ETL to eliminate the metadata stuff?  Which would make this fine. Otherwise I suppose we'd want to leave it in there until that's run.

Reply:
Yep, migration is there to do just that: https://github.com/alldigitalrewards/rewardstack/pull/140/files#diff-e645a397a9b2fdd2abca850ed4d253d570f8f5e5d72c0dc76ee029dc0d8c6d28R16